src/main/scala/servers/Metals.scala

```diff
@@ -62,30 +63,38 @@ object Metals extends ScalaLanguageServer { server =>
   )
 
   override def postInitialization(client: ScalaLanguageClient, activeServer: ActiveServer): Unit = {
-    activeServer
-      .connection
-      .asInstanceOf[js.Dynamic]
+    val projectPath = activeServer.projectPath
+    val connection = activeServer.connection.asInstanceOf[js.Dynamic]
+
+    connection
       .onCustom("metals/status", { params: js.Dynamic =>
-        client.statusBarTile.innerHTML = params.text.toString
+        Atom.workspace.getActiveTextEditor().foreach { editor =>
+          if (client.isFileInProject(editor, projectPath)) {
+            client.statusBarTile.innerHTML = params.text.toString
+          }
+        }
       })
 
-    activeServer
-      .connection
-      .asInstanceOf[js.Dynamic]
+    Atom.asInstanceOf[js.Dynamic]
+      .views
+      .addViewProvider({ _.getViewClass }: js.Function1[HtmlView, Element])
+
+    connection
       .onCustom("metals/executeClientCommand", { params: js.Dynamic =>
         params.command.toString match {
           case "metals-logs-toggle" =>
             dispatchAtomCommand("console:toggle")
           case "metals-diagnostics-focus" =>
             dispatchAtomCommand("diagnostics:toggle-table")
-          case "metals-doctor-run" => {
+          case "metals-doctor-run" | "metals-doctor-reload" => {
```

Comment:
`metals-doctor-reload` is sent after every build imoprt, even if the user doesn't have the Doctor tab open. In vscode we ignore `doctor-reload` if the doctor webwiew does not exist https://github.com/scalameta/metals-vscode/blob/7969642212be207805825f9d10ed6d76fee33453/src/extension.ts#L230

Reply:
Thanks! I thought so, but I just couldn't get it right with the weird Atom API for this.. I think I fixed in 51b5e43.